x-pack/plugins/security_solution/server/lib/detection_engine/rules/find_rules.ts

```diff
@@ -12,7 +12,7 @@ export const getFilter = (filter: string | null | undefined) => {
   if (filter == null) {
     return `alert.attributes.alertTypeId: ${SIGNALS_ID}`;
   } else {
-    return `alert.attributes.alertTypeId: ${SIGNALS_ID} AND ${filter}`;
+    return `alert.attributes.alertTypeId: ${SIGNALS_ID} AND (${filter})`;
```

Comment:
adding parenthesis due to `OR` joins in KQL passed to alerts client find here -> https://github.com/elastic/kibana/pull/76398/files#diff-a8b6548cd6fceb9a3bb78453de9e2d17R44-R45